File: CppPractice/Project1/Project1/Leetcode86.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

#include <map>

#include <bitset>

using namespace std;


struct ListNode {
    int val;
    ListNode* next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode* next) : val(x), next(next) {}
};


class Solution {
public:
    ListNode* partition(ListNode* head, int x) {
        vector<int> record;
        vector<int> new_record;
        ListNode* cur = head;
        int list_ind = 0;
        while (cur)
        {
            record.push_back(cur->val);
            cur = cur->next;
        }

        for (int i = 0; i < record.size(); i++)
        {
            if (record[i]<x)
            {
                new_record.push_back(record[i]);
            }

        }
        for (int i = 0; i < record.size(); i++)
        {
            if (record[i] >= x)
            {
                new_record.push_back(record[i]);
            }

        }
        cur = head;
        while (cur)
        {
            cur->val = new_record[list_ind];
            list_ind++;
            cur = cur->next;
        }
        return head;

    }
};

File: Pythonleetcode/Leetcode86.py
class ListNode(object):
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution(object):
    def partition(self, head, x):
        """
        :type head: ListNode
        :type x: int
        :rtype: ListNode
        """
        right_dummy = ListNode()
        left_dummy = ListNode()
        r = right_dummy
        l = left_dummy
        pointer = head
        while pointer != None:
            temp=ListNode(pointer.val)
            if pointer.val < x:
                l.next=temp
                l=l.next
            else:
                r.next = temp
                r = r.next
            pointer = pointer.next
            l.next = right_dummy.next

        return left_dummy.next